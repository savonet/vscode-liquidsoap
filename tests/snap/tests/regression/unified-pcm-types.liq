# Type parameters below are shared between pcm_s16 source and the decoded
# source. We want to make sure that this works at runtime.
def f() =
  s = blank()

  s = source.on_track(s, fun (_) -> test.pass())

  s = audio.decode.pcm_s16(s)

  output.file(
    %ffmpeg(
      %audio(codec="aac")
    ),
    "unified-pcm-types.aac",
    s
  )
end

test.check(f)
