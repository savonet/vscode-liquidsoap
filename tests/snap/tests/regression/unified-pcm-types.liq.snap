># Type parameters below are shared between pcm_s16 source and the decoded
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># source. We want to make sure that this works at runtime.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def f() =
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^ source.liquidsoap
>  s = blank()
#^^^^^^^^^^^^^^ source.liquidsoap
>
>  s = source.on_track(s, fun (_) -> test.pass())
#^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                         ^^^ source.liquidsoap keyword.control.liquidsoap
#                            ^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>
>  s = audio.decode.pcm_s16(s)
#^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                        ^^ source.liquidsoap constant.numeric.integer
#                          ^^^^ source.liquidsoap
>
>  output.file(
#^^^^^^^^^^^^^^^ source.liquidsoap
>    %ffmpeg(
#^^^^^^^^^^^^^ source.liquidsoap
>      %audio(codec="aac")
#^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                   ^ source.liquidsoap string.quoted.double.liquidsoap
#                    ^^^ source.liquidsoap string.quoted.double.liquidsoap
#                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                        ^^ source.liquidsoap
>    ),
#^^^^^^^ source.liquidsoap
>    "unified-pcm-types.aac",
#^^^^ source.liquidsoap
#    ^ source.liquidsoap string.quoted.double.liquidsoap
#     ^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                          ^ source.liquidsoap string.quoted.double.liquidsoap
#                           ^^ source.liquidsoap
>    s
#^^^^^^ source.liquidsoap
>  )
#^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>test.check(f)
#^^^^^^^^^^^^^^ source.liquidsoap
>